miscellaneous_scripts/extract_apt_patch_data.py

```diff
@@ -0,0 +1,172 @@
+import subprocess
+import argparse
+import json
+
+from enum import Enum
+
+class ModeType(str, Enum):
+    GENERATE = 'generate'
+    MODIFY = 'modify'
+
+RETURN_CODE_OK = [0]
+
+def list_of_strings(arg):
+    return arg.split(",") if arg else []
+
+
+def get_package_list_using_command(run_command="apt list --installed"):
+    """
+    Uses the input command to retrieve the list of installed/upgradable apt packages.
+    :param run_command: str, the input apt command
+    :return: list, the list of packages
+    """
+    otpt = subprocess.run(run_command, shell=True, capture_output=True, text=True, check=True)
+    result = otpt.stdout.strip().split("\n")
+    return [output_line.split("/")[0] for output_line in result if "/" in output_line]
+
+
+def get_installed_version_for_packages(package_list = []):
+    """
+    Finds the currently installed version of the packages.
+
+    :param package_list: list[str], List of packages
+    :return: dict[str, str], Dict with (keys=package names) and (values=installed package versions)
+    """
+    run_command="apt list --installed"
+    otpt = subprocess.run(run_command, shell=True, capture_output=True, text=True, check=True)
```

Comment:
Please use full words for variable names, or use common abbreviations, for code readability.

Reply:
Thanks, changed the name